src/HotChocolate/Core/src/Types/Types/Contracts/IInputFieldInfo.cs

```diff
@@ -23,4 +23,14 @@ public interface IInputFieldInfo : IHasName, IHasFieldCoordinate, IHasRuntimeTyp
     /// Gets a formatter that shall intercept deserialized values and reformat them.
     /// </summary>
     IInputValueFormatter? Formatter { get; }
+
+     /// <summary>
+     /// Defines if this field is deprecated.
+     /// </summary>
+     bool IsDeprecated { get; }
```

Comment:
We should not put this on the info ... but on the IInputField ... the deprecation is not relevant for serializers and stuff. The info is used by serialization for instance. The deprecation is only important as a user information and has no real value here.

Reply:
Moving it up will also lessen the impact.